Scripts/Runtime/WitAuthUtility.cs

```diff
@@ -6,18 +6,44 @@
  */
 
 using System;
+using System.Runtime.CompilerServices;
 #if UNITY_EDITOR
 using UnityEditor;
+using UnityEngine;
+
 #endif
 
 public class WitAuthUtility
 {
     public static bool IsIDETokenValid => Tokens.Length == 3;
+
+    private static string ideToken;
     public static string IDEToken
     {
 #if UNITY_EDITOR
-        get { return EditorPrefs.GetString("Wit::IDEToken", ""); }
-        set { EditorPrefs.SetString("Wit::IDEToken", value); }
+        get
+        {
+            if (null == ideToken)
+            {
+                try
+                {
+                    ideToken = EditorPrefs.GetString("Wit::IDEToken", "");
+                }
+                catch (Exception e)
+                {
+                    // This will happen if we don't prime the ide token on the main thread and
+                    // we access the server token editorpref value in a request.
+                    Debug.LogError(e.Message);
+                }
+            }
+
+            return ideToken;
+        }
+        set
+        {
+            ideToken = value;
+            EditorPrefs.SetString("Wit::IDEToken", ideToken);
+        }
```

Comment:
should we clean this up?

Reply:
Not yet, it is there for backwards compatibility.